src/EventSubscriber/ApiKeySubscriber.php

```
@@ -30,7 +38,16 @@ public function onKernelRequest(RequestEvent $event): void
         }
 
         $request = $event->getRequest();
+
+        if (in_array($request->getPathInfo(), $this->excludedPaths, true)) {
+            return;
+        }
+
         $providedKey = $request->headers->get('X-API-Key');
+        if ($providedKey === null) {
+            $session = $request->getSession();
+            $providedKey = $session ? $session->get('api_key') : null;
```

PHPStan (2.1.15)

Ternary operator condition is always true.


(ternary.alwaysTrue)